dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/TCL1.mo

```diff
@@ -0,0 +1,46 @@
+within Dynawo.Electrical.Controls.Voltage;
+
+/*
+* Copyright (c) 2015-2019, RTE (http://www.rte-france.com)
+* See AUTHORS.txt
+* All rights reserved.
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source time domain simulation tool for power systems.
+*/
+
+model TCL1 "Tap Changer Lock (TCL)"
+  /* Lock tap changers when the voltage level goes below a predefined threshold
+     in order to avoid a voltage collapse */
+  extends BaseClasses.BaseTCL;
+
+  import Modelica.Constants;
+
+  public
+
+    parameter Types.VoltageModule UMin "Minimum voltage threshold before tap-changer locking";
+
+    Connectors.ImPin UMonitored (value (unit = "kV")) "Monitored voltage";
+
+  equation
+
+    // Check when the monitored voltage goes below UMin
+    when UMonitored.value < UMin then
+      UUnderMin = true;
+      tUnderUmin = time;
+    elsewhen UMonitored.value >= UMin and pre(UUnderMin) then
+      UUnderMin = false;
+      tUnderUmin = Constants.inf;
+    end when;
+
+annotation(preferredView = "text",
+    Documentation(info = "<html><head></head><body>This model will send a lock order to transformers with tap-changers in order that the tap is locked to its current step if the voltage becomes lower than a threshold on some nodes. Such a mechanism enables to avoid voltage collapse on situations where the tap actions become negative for the system stability.<div><br></div>The detailed tap-changer-lock behavior is explained in the following state diagram:
+
```

Comment:
Adapt the annotation for this specific model (one node controlled)

Reply:
done